### PR TITLE
Update release playbook

### PR DESCRIPTION
**Pull request by dkelmer**
- add a potentially real link to view if the artifacts got deleted
- delete step about being added to bazel-sheriffs -- I'm not actually sure I should've deleted this, but I'm pretty sure I'm not in the group and I was able to complete the steps so I suspect it's not needed
- I think the instructions should include what artifacts should be uploaded to the repository and what to do if they aren't, but I don't know what those instructions would be

### CODE REVIEW

File: docs/release-playbook.md [hunk 1]
```diff
@@ -122,16 +122,13 @@ Create candidates with the release.sh script.
...
-    repository.
+    "Unblock step" for the deployment step. 
+    
+    *   This will upload the release candidate binaries to GitHub and our 
+        apt-get repository. The github link is probably of the form:
+        https://releases.bazel.build/0.25.0/rc1/index.html
 
-    *   If you don't have the permission, ask one of the Buildkite org
-        admins to add you to the
-        [bazel-sheriffs](https://buildkite.com/bazel?team=bazel-sheriffs)
```
Comment (fweikert) [inline docs/release-playbook.md]: Please keep this paragraph, but change the group to https://buildkite.com/organizations/bazel-trusted/teams/release-managers/members.

Membership is still required (you're already a member of that group).

Comment (fweikert) [reply docs/release-playbook.md]: Context: we moved the release pipeline to a separate Buildkite org (bazel-trusted), which required the creation of a new group.

Comment (philwo) [reply docs/release-playbook.md]: @dkelmer Friendly ping :)

Comment (dkelmer) [reply docs/release-playbook.md]: Sorry, this totally fell off my radar. Addressed the changes!

**Comment by philwo** [comment]
LGTM from my side.

@fweikert You're the expert on this, can you please review and merge this when you have a minute? :)

**Comment by fweikert** [comment]
LGTM, thank you!

**Pull request merged**